tensorflow/python/keras/layers/convolutional.py

```diff
@@ -426,24 +426,43 @@ def __init__(self,
                kernel_constraint=None,
                bias_constraint=None,
                **kwargs):
-    super(Conv1D, self).__init__(
-        rank=1,
-        filters=filters,
-        kernel_size=kernel_size,
-        strides=strides,
-        padding=padding,
-        data_format=data_format,
-        dilation_rate=dilation_rate,
-        activation=activations.get(activation),
-        use_bias=use_bias,
-        kernel_initializer=initializers.get(kernel_initializer),
-        bias_initializer=initializers.get(bias_initializer),
-        kernel_regularizer=regularizers.get(kernel_regularizer),
-        bias_regularizer=regularizers.get(bias_regularizer),
-        activity_regularizer=regularizers.get(activity_regularizer),
-        kernel_constraint=constraints.get(kernel_constraint),
-        bias_constraint=constraints.get(bias_constraint),
-        **kwargs)
+    rv=self.checkProperKernel(kernel_size)
+    if(rv==0):
+      super(Conv1D, self).__init__(
+          rank=1,
+          filters=filters,
+          kernel_size=kernel_size,
+          strides=strides,
+          padding=padding,
+          data_format=data_format,
+          dilation_rate=dilation_rate,
+          activation=activations.get(activation),
+          use_bias=use_bias,
+          kernel_initializer=initializers.get(kernel_initializer),
+          bias_initializer=initializers.get(bias_initializer),
+          kernel_regularizer=regularizers.get(kernel_regularizer),
+          bias_regularizer=regularizers.get(bias_regularizer),
+          activity_regularizer=regularizers.get(activity_regularizer),
+          kernel_constraint=constraints.get(kernel_constraint),
+          bias_constraint=constraints.get(bias_constraint),
+          **kwargs)
+        
+    
+        
+  def checkProperKernel(self,kernel_size):
+        if(isinstance(kernel_size,tuple)):
+    	    for i in kernel_size:
+    		    if(i==0):
+    			    raise ValueError("Kernel dimension cannot be zero")
+        else:
+    	    if(kernel_size==0):
+    		    raise ValueError("Kernel size cannot be zero")
+        return 0
```

Comment:
this is a very weird return value....the previous implementation looks mostly good, simply just check without returning anything

Reply:
Can I use some other return value like("True")  as some kind of check needs to be performed before super(Conv1D) is called? Else the super() function will be called at any cost and the problem addressed in #37334 will still exist.